lib/icedfrisby.js

```diff
@@ -701,6 +647,44 @@ class Frisby {
   }
 
 
+  /**
+   * @param {object} options object
+   *
+   * @desc Set configuration options on this object.
+   * inspectOnFailure: This is a really neat option that will help you figure out what is happening with your requests. Dumps request/response information to the logs.
+   * timeout: Sets the maximum time we'll wait for a response before failing the request
+   * retry: Number of times we'll try this request before returning a failure. If timeout is set, each retry uses the timeout.
+   * request: Options for the request module. An object containing any of these: https://github.com/request/request#requestoptions-callback
+   * json: Sets body to JSON representation of value and adds Content-type: application/json header. Additionally, parses the response body as JSON.
+  */
+  config (opts) {
+    if ('inspectOnFailure' in opts) {
+      this._inspectOnFailure = opts.inspectOnFailure
+    }
+    if ('timeout' in opts){
+      this._timeout = opts.timeout
+    }
+    if ('retry' in opts){
+      this._attempts.maxRetries = opts.retry
+    }
+    if ('request' in opts) {
+      for(const request_option in opts.request){
+        if(opts.request.hasOwnProperty(request_option)){
+          let value = opts.request[request_option]
+          if(typeof opts.request[request_option] === "object"){
+            value = JSON.parse(JSON.stringify(opts.request[request_option])) //break reference to previous object
```

Comment:
I would probably just use lodash's clone to eliminate any possible bugs with objects that can't be stringified back and forth like `Date`. I'm not sure that this would actually cause any issues in practice though.

Reply:
I've switched to lodash, but I've also rewritten this bit. When testing the rewrite I discovered a bug to do with header case-sensitivity which I've added a unit test for then solved. It's not as clean as I'd like, having that "special case" there, but I can't see a better way.